src/main/java/uk/nhs/hee/tis/trainee/credentials/service/CredentialMetadataService.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2023 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.credentials.service;
+
+import static java.util.stream.Collectors.groupingBy;
+
+import com.amazonaws.xray.spring.aop.XRayEnabled;
+import java.util.Comparator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.nhs.hee.tis.trainee.credentials.dto.CredentialType;
+import uk.nhs.hee.tis.trainee.credentials.model.CredentialMetadata;
+import uk.nhs.hee.tis.trainee.credentials.repository.CredentialMetadataRepository;
+
+/**
+ * A service for handling credentials.
+ */
+@Slf4j
+@Service
+@XRayEnabled
+public class CredentialMetadataService {
+
+  private final CredentialMetadataRepository credentialMetadataRepository;
+
+  /**
+   * Constructs a new CredentialMetadataService with the provided CredentialMetadataRepository.
+   *
+   * @param credentialMetadataRepository The repository used for managing CredentialMetadata.
+   */
+  CredentialMetadataService(CredentialMetadataRepository credentialMetadataRepository) {
+    this.credentialMetadataRepository = credentialMetadataRepository;
+  }
+
+  /**
+   * A method to get and return the latest credential by issuedAt time.
+   */
+  public List<CredentialMetadata> getTheLatestCredential(CredentialType type, String traineeId) {
```

Comment:
I wonder if we could give this a clearer name? As I understand it, it will return a list of the latest credentials issued against each of the given trainee's placements or programme memberships. To me, the name suggests a single credential will be returned. Maybe something like `getLatestCredentialsForPlacementsOrProgrammes` or `getLatestCredentialsForType`?